pkg/fleet-manager/restore_controller.go

```diff
@@ -0,0 +1,174 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	backupapi "kurator.dev/kurator/pkg/apis/backups/v1alpha1"
+)
+
+// RestoreManager reconciles a Restore object
+type RestoreManager struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RestoreManager) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&backupapi.Restore{}).
+		WithOptions(options).
+		Complete(r)
+}
+
+func (r *RestoreManager) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("restore", req.NamespacedName)
+
+	restore := &backupapi.Restore{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, restore); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("restore does not exist")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	patchHelper, err := patch.NewHelper(restore, r.Client)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to init patch helper for restore %s", req.NamespacedName)
+	}
+	defer func() {
+		if err := patchHelper.Patch(ctx, restore); err != nil {
+			reterr = utilerrors.NewAggregate([]error{reterr, errors.Wrapf(err, "failed to patch %s  %s", restore.Name, req.NamespacedName)})
+		}
+	}()
+
+	if !controllerutil.ContainsFinalizer(restore, RestoreFinalizer) {
+		controllerutil.AddFinalizer(restore, RestoreFinalizer)
+	}
+
+	// Handle deletion
+	if restore.GetDeletionTimestamp() != nil {
+		return r.reconcileDeleteRestore(ctx, restore)
+	}
+
+	// Handle the main reconcile logic
+	return r.reconcileRestore(ctx, restore)
+}
+
+// reconcileRestore handles the main reconcile logic for a Restore object.
+func (r *RestoreManager) reconcileRestore(ctx context.Context, restore *backupapi.Restore) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	fleetName, destinationClusters, err := r.fetchRestoreDestinationClusters(ctx, restore)
+	if err != nil {
+		log.Error(err, "failed to fetch destination clusters for restore", "restoreName", restore.Name)
```

Comment:
nit: donot you have key values in L50 	`log := ctrl.LoggerFrom(ctx).WithValues("restore", req.NamespacedName)`

Reply:
yes， forgot it